### ISSUE TITLE
Inference Script

### ISSUE THREAD

**Issue by rhodriguerrier** [DualPM/DualPM_Paper#2]
When will the inference script be available? Also, the link to feature extraction code leads to a 404 error, will this be fixed soon too?

Thanks

**Comment by ben-kaye** [comment]
Hi sorry for the delay. I have created an inference script. The feature extractor is a thin wrapper on [A Tale of 2 features](https://arxiv.org/abs/2305.15347) / [github](https://github.com/Junyi42/sd-dino). We use a 64 dim PCA trained on 1000 masked samples from the male horse in the dataset, masked at an output resolution 60x60. Annoyingly it is stuck in python 3.9 because of ODISE. It should be available at https://github.com/tomasjakab/canon/tree/refactor when published

**Comment by ben-kaye** [comment]
https://github.com/DualPM/dualpm_features for generating features
https://huggingface.co/benkaye/dualpm for inference/weights
use `scripts/infer.py` with the weights_path of the relevant config pointing to the pytorch_model.bin downloaded from hugging face. or alternatively use the `recon, canon, confidence, occupancy = model.infer(feat, mask, return_pointclouds: bool)` with the hugging face from pretrained
